lib/puppet/type/archive.rb

```diff
@@ -150,6 +150,16 @@ def should_to_s(value)
     desc 'extract command group (using this option will configure the archive file permisison to 0644 so the user can read the file).'
   end
 
+  newparam(:proxy_type) do
+    desc 'proxy type (none|ftp|http|https)'
+    newvalues(:none, :ftp, :http, :https)
+    defaultto(:none)
+  end
+
+  newparam(:proxy_server) do
+    desc 'proxy address to use when accessing source'
+  end
+
```

Comment:
@alexjfisher, I believe so.

We are assuming if `proxy_server=https://foo_proxy` we are using `https` proxy_type, and we don't care if you are trying to download a file from `http://somefile`. Also the user can be explicit about the proxy_type and we won't guess.